src/Service/OpenCatalogiService.php

```diff
@@ -115,64 +115,87 @@ public function setConfiguration(array $configuration): void
             && $opencatalogi['logo'] === null
         ) {
-            $organization->hydrate(['logo' => null]);
+            $logo = null;
         }
 
         // If we get an empty string we set the logo from the github api.
         if (key_exists('logo', $opencatalogi) === true
             && $opencatalogi['logo'] === ''
         ) {
-            $organization->hydrate(['logo' => $organizationArray['avatar_url']]);
+            $logo = $organizationArray['avatar_url'];
         }
 
         // If we don't get a opencatalogi logo we set the logo from the github api.
         if (key_exists('logo', $opencatalogi) === false) {
-            $organization->hydrate(['logo' => $organizationArray['avatar_url']]);
+            $logo = $organizationArray['avatar_url'];
+        }
+
+        // If the logo is set hydrate the logo.
+        if (isset($logo) === true) {
+            $organization->hydrate(['logo' => $logo]);
+            $this->entityManager->persist($organization);
         }
 
+        return $organization;
+    }
+
+    /**
+     * This function enriches the opencatalogi file organization.
+     *
+     * @param array $organizationArray The organization from the github api.
+     * @param array $opencatalogi The opencatalogi file as array.
+     * @param ObjectEntity $organization The organization object.
+     *
+     * @return ObjectEntity The organization object with updated logo.
+     */
+    public function enrichDescription(array $organizationArray, array $opencatalogi, ObjectEntity $organization): ObjectEntity
+    {
         // If the opencatalogi description is set to null or false we set the organization description to null.
         if (key_exists('description', $opencatalogi) === true
             && $opencatalogi['description'] === false
             || key_exists('description', $opencatalogi) === true
             && $opencatalogi['description'] === null
         ) {
-            $organization->hydrate(['description' => null]);
+            $description = null;
         }
 
         // If we get an empty string we set the description from the github api.
         if (key_exists('description', $opencatalogi) === true
             && $opencatalogi['description'] === ''
         ) {
-            $organization->hydrate(['description' => $organizationArray['description']]);
+            $description = $organizationArray['description'];
         }
 
         // If we don't get a opencatalogi description we set the description from the github api.
         if (key_exists('description', $opencatalogi) === false) {
-            $organization->hydrate(['description' => $organizationArray['description']]);
+            $description = $organizationArray['description'];
         }
 
-        return $organization;
+        // If the description is set hydrate the description.
+        if (isset($description) === true) {
+            $organization->hydrate(['description' => $description]);
```

Comment:
This is better than before, but for in the future:
$organizationData = ['logo' => $logo, 'description' => $description];
$organization->hydrate($organizationData);

Reply:
Just a question here: where do you set the logo now? I'm not saying it's wrong, but I can't find it.

Reply:
I don't always want to set the description or logo if not matches the criteria. But I think I'm handling everything well, updated the service.

@rjzondervan I created an enrichLogo and enrichDescription where I hydrate the organization.